### ISSUE TITLE
Xcode tests crash

### ISSUE THREAD

**Issue by darkobacicf** [microsoft/azure-pipelines-tasks#15657]
**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: Xcode@5  

## Environment
- Server - Azure Pipelines or TFS on-premises?
   Azure Pipelines

- Agent - Hosted or Private: 
    
    - If using Hosted agent, provide agent queue name:

    - If using private agent, provide the OS of the machine running the agent and the agent version: 

## Issue Description
Hi,
we have tests for our iOS app that we want to run in the pipeline and publish the results. So far we tried everything but no luck! We always get the same error which you can see down below. We are running the tests on a simulator.

We run the tests using this command and always get **failed with return code: 65**: 

```
  # Unit Test
  - task: Xcode@5
    displayName: Run Unit tests
    continueOnError: true
    inputs:
      xcodeVersion: specifyPath
      xcodeDeveloperDir: /Applications/Xcode_13.1.app/Contents/Developer
      scheme: $(buildScheme)
      sdk: $(buildSDK)
      configuration: $(buildConfiguration)
      xcWorkspacePath: $(build.sourcesDirectory)/$(xcWorkspaceName)
      actions: test-without-building
      useXcpretty: true
      packageApp: false
      destinationTypeOption: simulators
      destinationPlatformOption: iOS
      destinationSimulators: $(destinationSimulators)
      publishJUnitResults: true
```

We even downloaded the xcresult files and tried to open them but we get this error: 

`Assertions: System: Early unexpected exit, operation never finished bootstrapping - no restart will be attempted. (Underlying Error: Crash: XXX (11968) <external symbol>: Namespace CODESIGNING, Code 0x2. dyld4 config: DYLD_ROOT_PATH=/Applications/Xcode_13.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot DYLD_LIBRARY_PATH=/Users/runner/Library/Developer/Xcode/DerivedData/XXX-dyverlevgsxcqeaeykrhtrlujmlh/Build/Products/Debug-iphonesimulator DYLD_INSERT_LIBRARIES=/Users/runner/Library/Developer/Xcode/DerivedData/XXX-dyverlevgsxcqeaeykrhtrlujmlh/Build/Products/Debug-iphonesimulator/XXX.app/Frameworks/libXCTestBundleInject.dylib DYLD_FRAMEWORK_PATH=/Users/runner/Library/Developer/Xcode/DerivedData/XXX-dyverlevgsxcqeaeykrhtrlujmlh/Build/Products/Debug-iphonesimulator. If you believe this error represents a bug, please attach the result bundle at /Users/runner/Library/Developer/Xcode/DerivedData/XXX-dyverlevgsxcqeaeykrhtrlujmlh/Logs/Test/TestXXX-2022.01.05_10-45-54-+0000.xcresult)`

We don't use AppCenter for now, maybe in the future. At this point I'm wondering if this is even supported since I can't find any guidelines/examples on your site for running tests in Azure Pipelines. The only [reference](https://docs.microsoft.com/en-us/azure/devops/pipelines/ecosystems/xcode?view=azure-devops#testing-on-azure-hosted-devices) you have is running the tests using **AppCenterTest@1**. While for Android you have cleared [steps](https://docs.microsoft.com/en-us/azure/devops/pipelines/ecosystems/android?view=azure-devops#test-on-the-android-emulator) how to do it in the Azure Pipeline itself.

### Error logs

```
2022-01-05 10:50:33.187 xcodebuild[9553:62818] [MT] IDETestOperationsObserverDebug: 141.005 elapsed -- Testing started completed.
2022-01-05 10:50:33.187 xcodebuild[9553:62818] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
2022-01-05 10:50:33.187 xcodebuild[9553:62818] [MT] IDETestOperationsObserverDebug: 141.005 sec, +141.005 sec -- end
Testing failed:
	XXXTests:
		XXX (11968) encountered an error (Early unexpected exit, operation never finished bootstrapping - no restart will be attempted. (Underlying Error: Crash: XXX (11968) <external symbol>: Namespace CODESIGNING, Code 0x2. dyld4 config: DYLD_ROOT_PATH=/Applications/Xcode_13.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot DYLD_LIBRARY_PATH=/Users/runner/Library/Developer/Xcode/DerivedData/XXX-dyverlevgsxcqeaeykrhtrlujmlh/Build/Products/Debug-iphonesimulator DYLD_INSERT_LIBRARIES=/Users/runner/Library/Developer/Xcode/DerivedData/XXX-dyverlevgsxcqeaeykrhtrlujmlh/Build/Products/Debug-iphonesimulator/XXX/Frameworks/libXCTestBundleInject.dylib DYLD_FRAMEWORK_PATH=/Users/runner/Library/Developer/Xcode/DerivedData/XXX-dyverlevgsxcqeaeykrhtrlujmlh/Build/Products/Debug-iphonesimulator. If you believe this error represents a bug, please attach the result bundle at /Users/runner/Library/Developer/Xcode/DerivedData/XXX-dyverlevgsxcqeaeykrhtrlujmlh/Logs/Test/Test-XXX-2022.01.05_10-45-54-+0000.xcresult))

** TEST FAILED **

##[error]Error: /usr/bin/xcodebuild failed with return code: 65
```

**Comment by anatolybolshakov** [comment]
Hi @darkobacicf sorry for the delay - Xcode task is the simple wrapper for xcodebuild cli tool - so if building of your project (including step with tests run) via xcodebuild call works fine with the same set of parameters, it should work.
If you suspect that there is some issue with xcodebuild - I would suggest to reach out to Xcode support for further troubleshooting.
Could you please also provide some minimal sample to reproduce the issue on DevOps side?

**Comment by anatolybolshakov** [comment]
Hi @darkobacicf do you have any updates?

**Comment by RafaelGruhn** [comment]
I have a similar problem when running the tests for IOS using the Mac runners from Azure DevOps, but I run the tests using the fastlane "scan" command:


```
2022-03-04 19:00:38.572 xcodebuild[8027:28537] [MT] IDETestOperationsObserverDebug: 432.726 elapsed -- Testing started completed.
2022-03-04 19:00:38.572 xcodebuild[8027:28537] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
2022-03-04 19:00:38.572 xcodebuild[8027:28537] [MT] IDETestOperationsObserverDebug: 432.726 sec, +432.726 sec -- end
Testing failed:
	XXXXXTests:
		XXXXXX (13275) encountered an error (Test runner never began executing tests after launching. If you believe this error represents a bug, please attach the result bundle at /Users/runner/Library/Developer/Xcode/DerivedData/XXXXX-atxlnkyaquusrnbybhrroennkhxn/Logs/Test/XXXXXXXXX-2022.03.04_18-29-16-+0000.xcresult)

** TEST FAILED **
Exit status: 65
```

**Comment by RafaelGruhn** [comment]
I solved this problem by installing version 14.5 of the IOS Simulator (by default I was using 15.0) and indicating for the tests to run on the simulators of that version

There seems to be an issue between the MAC version in Azure Pipelines and the version 15 simulators
